bindings/pyroot/pythonizations/python/ROOT/pythonization/_roofit/_roodataset.py

```diff
@@ -54,3 +54,200 @@ def plotOnXY(self, *args, **kwargs):
+            copy (bool): If False, the data will not be copied. Use with
+                         caution, as the numpy arrays and the RooAbsData now
+                         own the same memory. If the dataset uses a
+                         RooTreeDataStore, there will always be a copy and the
+                         copy argument is ignored.
+            compute_derived_weight (bool): Sometimes, the weight variable is
+                not stored in the dataset, but it is a derived variable like a
+                RooFormulaVar. If the compute_derived_weight is True, the
+                weights will be computed in this case and also stored in the
+                output. Switched off by default because the computation is
+                relatively expensive.
+
+        Returns:
+            dict: A dictionary with the variable or weight names as keys and
+                  the numpy arrays as values.
+        """
+        import ROOT
+        import numpy as np
+
+        data = {}
+
+        if isinstance(self.store(), ROOT.RooVectorDataStore):
+            for name, array in self.store().to_numpy(copy=copy).items():
+                data[name] = array
+        elif isinstance(self.store(), ROOT.RooTreeDataStore):
+            # first create a VectorDataStore so we can read arrays
+            store = self.store()
+            variables = store.get()
+            store_name = store.GetName()
+            tmp_store = ROOT.RooVectorDataStore(store, variables, store_name)
+            for name, array in tmp_store.to_numpy(copy=copy).items():
+                data[name] = array
+        else:
+            raise RuntimeError(
+                "Exporting RooDataSet to numpy arrays failed. The data store type "
+                + self.store().__class__.__name__
+                + " is not supported."
+            )
+
+        # Special case where the weight is a derived variable (e.g. a RooFormulaVar).
+        # We don't want to miss putting the weight in the output arrays, so we
+        # are forced to iterate over the dataset to compute the weight.
+        if compute_derived_weight:
+            try:
+                wgt_var_name = self.weightVar().GetName()
+                if not wgt_var_name in data:
+                    weight_array = np.zeros(self.numEntries(), dtype=np.float64)
+                    for i in range(self.numEntries()):
+                        self.get(i)
+                        weight_array[i] = self.weight()
+                    data[wgt_var_name] = weight_array
+            except ReferenceError:
+                # self.weightVar() is a nullptr
+                pass
+
+        return data
+
+    @staticmethod
+    def from_pandas(df, variables, name=None, title=None, weight_name=None, clip_to_limits=True):
+        """Create a RooDataSet from a pandas DataFrame.
+        Args:
+            df (pandas.DataFrame): Pandas DataFrame to import.
+            variables (RooArgSet): Specification of the variables in the
```

Comment:
This might also work for lists?

Reply:
That's true, I will specify that in the docs.